Dockerfile

```diff
@@ -27,7 +28,7 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
 # Build
 RUN --mount=type=cache,target=/root/.cache/go-build \
     --mount=type=cache,target=/go/pkg \
-    CGO_ENABLED=1 GOOS=$TARGETOS GOARCH=$TARGETARCH GO111MODULE=on go build -ldflags="-s -w" -a -o libvirt-provider ./provider/cmd/main.go
+    CGO_ENABLED=1 GOOS=$TARGETOS GOARCH=$TARGETARCH GO111MODULE=on go build -ldflags="-s -w" -a -o libvirt-provider ./cmd/libvirt-provider/main.go
```

Comment:
does -a some useful reason?

```
-a
	force rebuilding of packages that are already up-to-date.
```

Reply:
There are a couple of advantages with using `-a` when building the go binaries:

- When building binaries for a different target platform (OS/architecture) than the one you're currently working on, using -a can be beneficial to ensure all dependencies are correctly compiled for the target platform.
- If you've updated dependencies (e.g., using go get to fetch a newer version of a library) and want to ensure that these changes are fully incorporated into your binary, using -a forces a rebuild of these dependencies, taking into account the updated versions.
- After upgrading your Go compiler or the Go version itself, using -a ensures that all packages are rebuilt using the new compiler or Go version. This can help leverage optimizations or language features introduced in newer versions.

On your local box you don't need `-a` for development purposes - but it is safer in a build environment.